nixos/modules/services/network-filesystems/juicefs.nix

```diff
@@ -0,0 +1,200 @@
+        type = lib.types.attrsOf lib.types.str;
+        default = { };
+        example = {
+          endpoint = "minio.example.com:9000";
+        };
+        description = "Additional storage options (like endpoint for S3)";
+      };
+
+      # Cache options
+      cacheDir = lib.mkOption {
+        type = lib.types.str;
+        default = "/var/cache/juicefs";
+        description = "Directory for local cache";
+      };
+
+      cacheSize = lib.mkOption {
+        type = lib.types.str;
+        default = "100G";
+        description = "Size of local cache";
+      };
+
+      compress = lib.mkOption {
+        type = lib.types.enum [
+          "none"
+          "lz4"
+          "zstd"
+        ];
+        default = "none";
+        description = "Compression algorithm";
+      };
+
+      writeback = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = "Enable write back cache for better performance";
+      };
+
+      extraArgs = lib.mkOption {
+        type = lib.types.listOf lib.types.str;
+        default = [ ];
+        description = "Additional arguments for mount command";
+        example = [
+          "--no-verify-ssl"
+          "--cache-partial-only"
+        ];
+      };
+    };
+  };
+
+in
+{
+  options.services.juicefs = {
+    enable = lib.mkEnableOption "JuiceFS Service";
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.juicefs;
+      description = "JuiceFS package to use";
+    };
+
+    volumes = lib.mkOption {
+      type = lib.types.attrsOf volumeOptions;
+      default = { };
+      description = "Set of JuiceFS volumes to mount";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    # Create cache directories
+    systemd.tmpfiles.rules = lib.concatLists (
+      lib.mapAttrsToList (name: volume: [
+        "d ${volume.cacheDir} 0755 root root -"
+      ]) cfg.volumes
+    );
+
+    systemd.services = lib.mapAttrs' (
+      name: volume:
+      lib.nameValuePair "juicefs-${name}" {
+        description = "JuiceFS volume ${name}";
+
+        wantedBy = [ "multi-user.target" ];
+        wants = [ "network-online.target" ];
+        requires = [ "network-online.target" ];
+        after = [ "network-online.target" ];
+
+        serviceConfig = {
+          Type = "forking";
+
+          # Environment variables for credentials and settings
+          Environment =
+            [
+              "JFS_NO_UPDATE_FSTAB=1"
+              "AWS_ENDPOINT=${volume.storageOptions.endpoint or ""}"
+            ]
+            ++ lib.optional (volume.accessKey != "") "ACCESS_KEY=${volume.accessKey}"
+            ++ lib.optional (volume.secretKey != "") "SECRET_KEY=${volume.secretKey}";
+
+          # Initialize the volume if needed
+          ExecStartPre = pkgs.writeShellScript "juicefs-init-${name}" ''
+            set -e
+            export PATH="${pkgs.coreutils}/bin:$PATH"
```

Comment:
Place this in the `path` option of the service. e.g. `systemd.service."foo".path = [ pkgs.coreutils ];

But aside from that, what part of coreutils do you need, and I'm not mistaken coreutils are already in the PATH of every unit?